packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -395,6 +395,20 @@ export function renderWithHooks<Props, SecondArg>(
       current !== null && current.type !== workInProgress.type;
   }
 
+  // Reset the memoCache index and create a backup copy in case of a setState during render
+  // or error, either of which can leave the cache in an inconsistent state
+  let previousMemoCache = null;
+  if (enableUseMemoCacheHook) {
+    previousMemoCache = workInProgress.memoCache;
+    if (previousMemoCache !== null) {
+      const memoCache = {
+        data: previousMemoCache.data.map(array => array.slice()),
+        index: 0,
+      };
+      workInProgress.memoCache = memoCache;
+    }
+  }
+
   workInProgress.memoizedState = null;
   workInProgress.updateQueue = null;
```

Comment:
@acdlite `updateQueue` is set to null here, so i'm not sure how to pass the memo cache via that object.

Reply:
It's on the current fiber: `current.updateQueue`. This is a really common pattern in the reconciler — conceptually the whole render phase is a copy-on-write of the "current" tree, where "work-in-progress" is the copy.

Reply:
`current` isn't in scope in this file, how do i access this? you mentioned elsewhere that we try not to access `.alternate`, so it seems like i shouldn't do `currentlyRenderingFiber.alternate`.